std/traits.d

```diff
@@ -2518,6 +2519,69 @@ private:
     }
 }
 
+/**
+ * Get an alias to a specific overload of a method.
+ *
+ * Params:
+ *    overloadSet = alias to the method's overload set
+ *    attrs = (optional) $(LREF FunctionAttribute)s of the overload to be aliased.
+ *        This is matched exactly! Note that one of @system, @trusted and @safe
+ *        is always there. Defaults to `FunctionAttribute.system`.
+ *    ParamTypes = Types of the parameters of the overload to be aliased.
+ */
+template getOverload(alias overloadSet, FunctionAttribute attrs, ParamTypes...)
+{
+    import std.meta : Filter;
+
+    private enum bool isMatch(alias fun) = (functionAttributes!fun == attrs
+        && is(ParamTypes == Parameters!fun));
+
+    private alias matches = Filter!(isMatch, __traits(getOverloads,
```

Comment:
the `getOverloads` compiler trait can take an optional `bool` parameter that indicates if the templates are included in the results. I wonder if, using this parameter, the template could also instantiate an overload, which would make it more versatile.

Reply:
Interesting suggestion, but I don't think it is feasible. ˋParamTypesˋ just isn't enough information to instantiate the overload.